client/src/components/user/ApiKey.jsx

```diff
@@ -0,0 +1,172 @@
+                    updated.
+                </NoteBlock>
+            )}
+            <div>
+                {apiKey && (
+                    <div>
+                        <NoteBlock
+                            header={<b>API key generated</b>}
+                            icon="check_circle"
+                        >
+                            Make sure to copy your new API key below. You won’t
+                            be able to see it again!
+                            <div
+                                style={{
+                                    display: 'flex',
+                                    alignItems: 'center',
+                                }}
+                            >
+                                <code>{apiKey}</code>
+                                <IconButton
+                                    tooltip="Copy API key"
+                                    onClick={handleCopyToClipboard}
+                                >
+                                    <FontIcon className="material-icons">
+                                        content_paste
+                                    </FontIcon>
+                                </IconButton>
+                            </div>
+                        </NoteBlock>
+                    </div>
+                )}
+                <div style={styles.flexCenterDiv}>
+                    API is key active. API key was generated at{' '}
+                    {new Date(createdAt).toLocaleString()}
+                    <FlatButton
+                        label="Delete API key"
+                        onClick={onDelete}
+                        style={{ color: 'red' }}
+                        icon={isUpdating ? <Spinner inButton /> : null}
+                        disabled={isUpdating}
+                    />
+                </div>
+            </div>
+        </div>
+    )
+}
+
+const ApiKeyStatus = () => {
+    const [apiKey, setApiKey] = useState(null)
+    const [isUpdating, setIsUpdating] = useState(false)
+    const { data, error, mutate } = useQuery('key', null, requestOpts)
+
+    const handleDeleteKey = async () => {
+        setIsUpdating(true)
+        await deleteApiKey()
+        setIsUpdating(false)
+        mutate({ hasApiKey: false, createdAt: undefined })
+    }
+
+    const handleGenerateKey = async () => {
+        try {
+            setIsUpdating(true)
+            const { apiKey } = await generateApiKey()
+            setIsUpdating(false)
+            setApiKey(apiKey)
+            mutate({ hasApiKey: true, createdAt: new Date() }, false)
+        } catch (e) {
+            console.error(e)
+            //TODO: show snackbar?
+        }
+    }
+
+    if (error) {
+        return 'Failed to load API-key status'
+    }
+    if (!data) {
+        return <Spinner />
+    }
+
+    return data.hasApiKey ? (
+        <ApiKeyDisplay
+            createdAt={data.createdAt}
```

Comment:
`data.createdAt` is inserted by the client in `handleGenerateKey` and is not present in the server response. This means that if users refresh the page, they see `API key was generated at Invalid Date`.

My suggestion is to add a `created_at` column to the `user_api_key` table and to change `getApiKeyByUserId` in `services/apiKey.js` so that it returns `{ hash: "...", createdAt: "..." }`.